File: VisualBasic/Hanoi/hanoi.vb
Imports System

Public Class Hanoi
  Public Sub HanoiExec(n As Integer, from As String, to_ As String, via As String)
    If n > 1 Then
      Call HanoiExec(n - 1, from, via, to_)
      Console.WriteLine(from & " -> " & to_)
      Call HanoiExec(n - 1, via, to_, from)
    Else
      Console.WriteLine(from & " -> " & to_)
    End If
  End Sub

  Public Shared Sub Main()
    Dim n As Integer
    n = Console.ReadLine()
    Dim h As New Hanoi
    h.HanoiExec(n, "a", "b", "c")
  End Sub
End Class

File: Objective-C/Hanoi/hanoi.m
#import <stdio.h>

void hanoi(int n, char from, char to, char via) {
  if (n > 1) {
    hanoi(n - 1, from,  via, to);
    printf("%c -> %c\n", from, to);
    hanoi(n - 1, via, to, from);
  } else {
    printf("%c -> %c\n", from, to);
  }
}

int main() {
  int n;
  scanf("%d", &n);

  hanoi(n, 'a', 'b', 'c');

  return 0;
}
